packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts

```diff
@@ -58,25 +58,41 @@ export class OdspDriverUrlResolver implements IUrlResolver {
             url = url.substr(1);
         }
         const odspResolvedUrl = resolvedUrl as IOdspResolvedUrl;
-        return `${odspResolvedUrl.siteUrl}/${url}?driveId=${encodeURIComponent(
-            odspResolvedUrl.driveId)}&itemId=${encodeURIComponent(
-                odspResolvedUrl.itemId,
-            )}&path=${encodeURIComponent("/")}`;
+        let odspUrl = `${odspResolvedUrl.siteUrl}/${url}?driveId=${encodeURIComponent(odspResolvedUrl.driveId,
+            )}&itemId=${encodeURIComponent(odspResolvedUrl.itemId)}&path=${encodeURIComponent("/")}`;
+        if (odspResolvedUrl.codeHint?.containerPackageName) {
+            odspUrl += `&containerPackageName=${encodeURIComponent(odspResolvedUrl.codeHint?.containerPackageName)}`;
+        }
+
+        return odspUrl;
     }
 
-    public createCreateNewRequest(siteUrl: string, driveId: string, filePath: string, fileName: string): IRequest {
-        return createOdspCreateContainerRequest(siteUrl, driveId, filePath, fileName);
+    public createCreateNewRequest(
+        siteUrl: string,
+        driveId: string,
+        filePath: string,
+        fileName: string,
+        containerPackageName?: string,
```

Comment:
This doesn't make sense to me, we already have access to IFluidCodeDetails/quorum when in the create new process, why would we need a different param for this?

Reply:
In  create new we will extract package name from quorum while attaching and add header in request. Then while resolving we will add it to resolved url from request headers.

Reply:
Yes, here it should not be passed from host. We should remove this,

Reply:
> Yes, here it should not be passed from host. We should remove this, @abrahamrq You can remove it from here and just assume it will be in headers for create new. I will do a PR for extracting it from quorum and add it in request headers.

Cool Thanks